src/components/ModalWithForm/ModalWithForm.js

```diff
@@ -13,6 +14,31 @@ export default function ModalWithForm({
   signinError,
   signupError,
 }) {
+  // when press 'esc' close popup
+  useEffect(() => {
```

Comment:
* when I open a popup, then close it with `Escape` and press `Escape` again, the popup opens


please, add ` if (!isOpen) return;` to prevent adding the listener when the popup is closed


```js
  useEffect(() => {

    if (!isOpen) return; // prevent adding the listener when the popup is closed

    const exitEsc = (e) => {
      if (e.key === "Escape") {
        onClose();
      }
    };
    document.addEventListener("keydown", exitEsc);

    return () => document.removeEventListener("keydown", exitEsc);
  }, [isOpen, onClose]);


```

Reply:
Thank you. I fixed it.